core/field_variable.js

```diff
@@ -96,39 +122,79 @@ Blockly.FieldVariable.prototype.setSourceBlock = function(block) {
 };
 
 /**
- * Get the variable's name (use a variableDB to convert into a real name).
- * Unline a regular dropdown, variables are literal and have no neutral value.
- * @return {string} Current text.
+ * Get the variable's ID.
+ * @return {string} Current variable's ID.
  */
 Blockly.FieldVariable.prototype.getValue = function() {
-  return this.getText();
+  return this.variable_ ? this.variable_.getId() : null;
 };
 
 /**
- * Set the variable name.
- * @param {string} value New text.
+ * Get the text from this field.
```

Comment:
Should we leave the part in the comment about this being a variable name as well?

Reply:
Done.